packages/datadog-instrumentations/src/mongodb-core.js

```diff
@@ -89,7 +95,7 @@ function wrapUnifiedCommand (command, operation, name) {
   return shimmer.wrap(command, wrapped)
 }
 
-function wrapConnectionCommand (command, operation, name) {
+function wrapConnectionCommand (command, operation, name, isPromise = false) {
```

Comment:
The boolean value is essentially a two-value enum for selecting possible function values here; you could just as well be passing the functions themselves and declare the parameter as `instrumentFn = instrument` and then invoke
```
instrumentFn(operation, command, this, arguments, topology, ns, ops, { name })
```
below instead of decoding the boolean into a function selection with an if/else statement. Finally, you'd also pass `instrumentPromise` instead of `true` in the >=6.4.0 addHook invocation.

This is admittedly a matter of style, but I obviously find the one with functions themselves being passed to be more concise and in fact easier to comprehend in call sites as the parameter values are now named as `instrument` and `instrumentPromise` instead of `false/`true`. For extra legibility, you could remove the default (if that's possible) and rename `instrument` to `instrumentCallback` so it'd be explicitly featured in the call site in addHook for < 6.4.0 case – again, even easier to comprehend the code when read.

Reply:
I like your approach more, I'll modify it